### ISSUE TITLE
Please catch your exceptions

### ISSUE THREAD

**Issue by kergoth** [buildinspace/peru#169]
I'm tired of seeing screenfuls of tons of stack traces every time there's a problem, and then wasting a ton of time digging through it to figure out what's wrong.

As one example:

```
  Traceback (most recent call last):
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/urllib/request.py", line 1254, in do_open
      h.request(req.get_method(), req.selector, req.data, headers)
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/http/client.py", line 1106, in request
      self._send_request(method, url, body, headers)
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/http/client.py", line 1151, in _send_request
      self.endheaders(body)
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/http/client.py", line 1102, in endheaders
      self._send_output(message_body)
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/http/client.py", line 934, in _send_output
      self.send(msg)
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/http/client.py", line 877, in send
      self.connect()
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/http/client.py", line 849, in connect
      (self.host,self.port), self.timeout, self.source_address)
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/socket.py", line 711, in create_connection
      raise err
    File "/opt/homebrew/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/socket.py", line 702, in create_connection
      sock.connect(sa)
  OSError: [Errno 65] No route to host
```

**Comment by oconnor663** [comment]
Could you include what version of peru you're running, and how you produced that error? We substantially changed the error handling in 1.0.1, which went out last month. `git` modules should be reasonably clean now, though I think our `curl` module type is still pretty spammy.

**Comment by kergoth** [comment]
Ah, glad to hear it's improving, thanks. But this is with 1.0.1.

**Comment by oconnor663** [comment]
@kergoth could you give me a way to repro this one?

**Comment by kergoth** [comment]
Another example where a failure exception isn't being caught, which makes it almost impossible to diagnose, since it doesn't tell me which url failed:
```
Traceback (most recent call last):
  File "/Users/kergoth/.local/share/virtualenvs/peru/lib/python3.6/site-packages/peru/resources/plugins/curl/curl_plugin.py", line 169, in <module>
    main()
  File "/Users/kergoth/.local/share/virtualenvs/peru/lib/python3.6/site-packages/peru/resources/plugins/curl/curl_plugin.py", line 163, in main
    plugin_reup(url, sha1)
  File "/Users/kergoth/.local/share/virtualenvs/peru/lib/python3.6/site-packages/peru/resources/plugins/curl/curl_plugin.py", line 150, in plugin_reup
    with urllib.request.urlopen(url) as request:
  File "/opt/homebrew/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/opt/homebrew/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/opt/homebrew/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/opt/homebrew/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/opt/homebrew/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/opt/homebrew/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found
peru reup failed
```

**Comment by oconnor663** [comment]
Thanks for reporting. It looks like when I put a nonexistent URL into a `curl` module, `peru sync` will print out the proper `In target ...:` line above the error. But `peru reup` doesn't. As part of cleaning up the unnecessary stack trace here, I'll make sure `reup` applies the right context.

**Comment by oconnor663** [comment]
Just pushed aa6fc754b8310201f4f3c240f4582721443d3599 and 967af6f048b66a0b478f69cee86fd031acd833b6 to help with this.

**Comment by oconnor663** [comment]
I'll need to fix some flake8 errors before I can get Travis to pass and then ship a new version to PyPI, but you should be able to test the new changes from master now if you have time.

**Comment by oconnor663** [comment]
This should be fixed now, and I've just released version 1.1.3. Apologies for the...extended incubation period.
